pkg/apis/core/validation/validation_test.go

```diff
@@ -15477,6 +15478,7 @@ func TestValidateServiceCreate(t *testing.T) {
 		name           string
 		tweakSvc       func(svc *core.Service) // given a basic valid service, each test case can customize it
 		numErrs        int
+		featureGates   []featuregate.Feature
```

Comment:
You probably started writing this before I pushed the PreferSameTrafficDistribution update that removed this field, right?

The problem is that having `featureGates` works during alpha when the feature is off by default and you're specifying gates that should be enabled, but then [it breaks in beta when the feature is on by default](https://github.com/kubernetes/kubernetes/pull/132127#issuecomment-2956921143). It works better to [just have a `bool` here for whether to enable this feature gate](https://github.com/kubernetes/kubernetes/compare/5f454bb407ccbd2f510150fed8cd37b4033b10e8..dacafec88c28e970f6064f4542e6008502188a83), like we do with the other two.

Reply:
> You probably started writing this before I pushed the PreferSameTrafficDistribution update that removed this field, right?

Yup! 

> The problem is that having `featureGates` works during alpha when the feature is off by default and you're specifying gates that should be enabled

Ah thanks, that makes sense.